Assets/Scripts/Ships/DataManagement/ShipData.cs

```diff
@@ -18,6 +19,7 @@ public class ShipData : UniqueId
         [SerializeField] private GameObject visuals;
         [SerializeField] private int cost;
         [SerializeField] private List<AttackScriptableObject> weapons;
+        [SerializeField] private List<ModuleList> moduleGrid;
```

Comment:
I don't think this is a good way storing module grids.

It seems like there are two pieces of information you might want to store for a ship class/type: the shape/size of the module grid & default modules. If we are assuming that all grids will be rectangular then a width/height might work, otherwise you'll need to do something like what you did for representing module sizes. I don't know if we should worry about default modules for ship yet, but a list of modules might be the simplest solution for that (ideally with a custom inspector to make placing them easier).